doc: update doc for 2.5.1 (#13341)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-2.5.1.rst

See :ref:`nrf9160_gs_updating_fw_modem` for instructions.

Modem-related libraries and versions
====================================

.. list-table:: Modem-related libraries and versions
   :widths: 15 10
   :header-rows: 1

   * - Library name
     - Version information
   * - Modem library
     - `Changelog <Modem library changelog for v2.5.1_>`_
   * - LwM2M carrier library
     - `Changelog <LwM2M carrier library changelog for v2.5.1_>`_

Known issues
************

Known issues are only tracked for the latest official release.
See `known issues for nRF Connect SDK v2.5.1`_ for the list of issues valid for the latest release.

.. _ncs_release_notes_251_changelog:

Changelog
*********

The following sections provide detailed lists of changes by component.

Protocols
=========

This section provides detailed lists of changes by :ref:`protocol <protocols>`.

Wi-Fi
-----

* Added improvements and fixes related to the following:

  * :term:`Delivery Traffic Indication Message (DTIM)` power save functionality
  * Firmware issues due to AP power cycle
  * MAC address configuration
  * General system stability

> 

Reviewer: Due to other changes, we had to adopted also some things on Matter, I would add those: 

Matter:
 - Fixed an issue with commissioning Matter over Wi-Fi devices to 5th fabric
 - Aligned the default values of attributes in ZAP files to comply with the Matter 1.1 specification
 - Fixed an issue with the Matter Bridge not being operational in some commercial ecosystems

Reviewer: The last one is on PR: https://github.com/nrfconnect/sdk-nrf/pull/13399
But hopefully can be merged today :)

nrfx
====

* Updated the nrfx QSPI driver behavior.
  Now, QSPI peripheral is not activated during the driver initialization but rather when the first operation is requested or when the :c:func:`nrfx_qspi_activate` function is called.
* Fixed an issue when external memory requires a relatively long time for the operation to happen.

>   This is done by adding a timeout member to :c:struct:`nrfx_qspi_config_t` , which can be configured at runtime.

Reviewer 2: ```suggestion
  This is done by adding a timeout member to :c:struct:`nrfx_qspi_config_t`, which can be configured at runtime.
```

  If the timeout value is ``0``, the old value is used instead (500 milliseconds).
* Removed the ongoing custom instruction long transfer check from the deactivate function.
  The check could trigger errors, and now you must make sure that the custom instruction long transfer has ended, otherwise it will be interrupted.

>   This change affects the behavior of the :c:func:`nrfx_qspi_uinit` and the :c:func:`nrfx_qspi_deactivate` functions.

Reviewer 2: ```suggestion
  This change affects the behavior of the :c:func:`nrfx_qspi_uinit` and :c:func:`nrfx_qspi_deactivate` functions.
```
Might be good to have a bit further information on how the change affects the behavior of the functions.

Reviewer 3: Maybe it's unclear but it's described above. The check for ongoing custom instruction has been removed so now when the user calls `nrfx_qspi_uninit` or `nrfx_qspi_deactivate` there is a chance that this will interrupt ongoing custom instruction transfer as it's not checked anymore

Reviewer 2: Right, at least it was not clear to me. Maybe the nrfx_qspi_deactivate function should be mentioned in the first sentence of the bullet item. The last sentence with the changed behavior of these two functions feels a bit off.

Author: I have rephrased the sentence a bit as per the above comment. Please check.

.. _ncs_release_notes_251:

|NCS| v2.5.1 Release Notes
##########################

.. contents::
   :local:
   :depth: 2

|NCS| delivers reference software and supporting libraries for developing low-power wireless applications with Nordic Semiconductor products in the nRF52, nRF53, nRF70, and nRF91 Series.
The SDK includes open source projects (TF-M, MCUboot, OpenThread, Matter, and the Zephyr RTOS), which are continuously integrated and redistributed with the SDK.

Release notes might refer to "experimental" support for features, which indicates that the feature is incomplete in functionality or verification, and can be expected to change in future releases.
To learn more, see :ref:`software_maturity`.

Highlights
**********

This patch release adds changes on top of the :ref:`nRF Connect SDK v2.5.0 <ncs_release_notes_250>`.

> The changes affect Matter, Wi-Fi, nrfx, Zephyr, and Trusted Firmware-M (TF-M).

Reviewer 4: ```suggestion
The changes affect Matter, Wi-Fi® , nrfx, Zephyr, and Trusted Firmware-M (TF-M).
```

.. _ncs_release_notes_251:

|NCS| v2.5.1 Release Notes
##########################

.. contents::
   :local:
   :depth: 2

|NCS| delivers reference software and supporting libraries for developing low-power wireless applications with Nordic Semiconductor products in the nRF52, nRF53, nRF70, and nRF91 Series.
The SDK includes open source projects (TF-M, MCUboot, OpenThread, Matter, and the Zephyr RTOS), which are continuously integrated and redistributed with the SDK.

Release notes might refer to "experimental" support for features, which indicates that the feature is incomplete in functionality or verification, and can be expected to change in future releases.
To learn more, see :ref:`software_maturity`.

Highlights
**********

> This patch release adds changes on top of the :ref:`nRF Connect SDK v2.5.0 <ncs_release_notes_250>`.
> 
> .. To do

Reviewer 5: ```suggestion
This patch release adds changes on top of the :ref:`nRF Connect SDK v2.5.0 <ncs_release_notes_250>`. The changes affect Matter, Wi-Fi, nrfx, Zephyr, and Trusted Firmware-M (TF-M).
```

.. _ncs_release_notes_251:

|NCS| v2.5.1 Release Notes
##########################

.. contents::
   :local:
   :depth: 2

|NCS| delivers reference software and supporting libraries for developing low-power wireless applications with Nordic Semiconductor products in the nRF52, nRF53, nRF70, and nRF91 Series.
The SDK includes open source projects (TF-M, MCUboot, OpenThread, Matter, and the Zephyr RTOS), which are continuously integrated and redistributed with the SDK.

Release notes might refer to "experimental" support for features, which indicates that the feature is incomplete in functionality or verification, and can be expected to change in future releases.
To learn more, see :ref:`software_maturity`.

Highlights
**********

This patch release adds changes on top of the :ref:`nRF Connect SDK v2.5.0 <ncs_release_notes_250>`.

> The changes effect Matter, `nrfx`_ v3.2.0, Wi-Fi, TF-M, :ref:`liblwm2m_carrier_readme` library, and modem firmware v1.3.6.

Reviewer 3: The nrfx that is integrated into this release is in 3.1.0 version

.. list-table:: Modem-related libraries and versions
   :widths: 15 10
   :header-rows: 1

   * - Library name
     - Version information
   * - Modem library
     - `Changelog <Modem library changelog for v2.5.1_>`_
   * - LwM2M carrier library
     - `Changelog <LwM2M carrier library changelog for v2.5.1_>`_

Known issues
************

Known issues are only tracked for the latest official release.
See `known issues for nRF Connect SDK v2.5.1`_ for the list of issues valid for the latest release.

.. _ncs_release_notes_251_changelog:

Changelog
*********

The following sections provide detailed lists of changes by component.

nrfx
====

* Updated the nrfx QSPI driver behavior.
  Now QSPI peripheral is not activated during the driver initialization but rather when the first operation is requested or when the :c:func:`nrfx_qspi_activate` function is called.
* Fixed an issue when external memory requires a relatively long time for the operation to happen.
  This is done by adding a timeout member to :c:struct:`nrfx_qspi_config_t` , which can be configured at runtime timeout.
  If the timeout value is ``0``, the old value is used instead (500 milliseconds).

> 

Reviewer 3: One additional commit went into the release https://github.com/nrfconnect/sdk-hal_nordic/commit/9784731461018d3e983604698fbbed6af2bea801

* Removed from the deactivate function check if there is ongoing custom instruction long transfer. The check could trigger anomalies and now it is the user's responsibility to be sure that the custom instruction long transfer has ended otherwise it will be interrupted. The change affects the behaviour of the `nrfx_qspi_uinit()` and the `nrfx_qspi_deactivate()` functions.

Author: updated. Please check.

Zephyr

> ======

Reviewer 6: An entry that mentions the workaround for anomaly 168 should be added somewhere here. Perhaps in a similar form as was used in [v2.1.4](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/2.4.2/nrf/releases/release-notes-2.1.4.html#zephyr).

Known issues
************

Known issues are only tracked for the latest official release.
See `known issues for nRF Connect SDK v2.5.1`_ for the list of issues valid for the latest release.

.. _ncs_release_notes_251_changelog:

Changelog
*********

The following sections provide detailed lists of changes by component.

nrfx
====

* Updated the nrfx QSPI driver behavior.
  Now QSPI peripheral is not activated during the driver initialization but rather when the first operation is requested or when the :c:func:`nrfx_qspi_activate` function is called.
* Fixed an issue when external memory requires a relatively long time for the operation to happen.
  This is done by adding a timeout member to :c:struct:`nrfx_qspi_config_t` , which can be configured at runtime timeout.
  If the timeout value is ``0``, the old value is used instead (500 milliseconds).

Zephyr
======

The Zephyr fork in |NCS| (``sdk-zephyr``) contains all commits from the upstream Zephyr repository up to and including ``a768a05e6205e415564226543cee67559d15b736``.
It also contains some |NCS| specific additions and commits cherry-picked from the upstream Zephyr repository including the following one:

> * Added workaround in ``sdk-zephyr`` for the Anomaly 160 that is affecting the nRF5340 SoC.

Reviewer 6: ```suggestion
* Added workaround in ``sdk-zephyr`` for the Anomaly 168 that is affecting the nRF5340 SoC.
```
